packages/autorest.python/autorest/codegen/serializers/model_serializer.py

```diff
@@ -72,16 +89,75 @@ def get_properties_to_initialize(model: ModelType) -> List[Property]:
             properties_to_initialize = model.properties
         return properties_to_initialize
 
+    @staticmethod
+    def get_properties_to_declare(model: ModelType) -> List[Property]:
+        if model.parents:
+            properties_to_declare = list(
+                {
+                    p.client_name: p
+                    for bm in model.parents
+                    for p in model.properties
+                    if p not in cast(ModelType, bm).properties
+                }.values()
+            )
+        else:
+            properties_to_declare = model.properties
+        return properties_to_declare
+
     def declare_model(self, model: ModelType) -> str:
         basename = (
             "msrest.serialization.Model"
             if self.code_model.options["client_side_validation"]
+            else "_model_base.Model"
+            if self.code_model.options["models_mode"] == "dpg"
             else "_serialization.Model"
         )
         if model.parents:
             basename = ", ".join([cast(ModelType, m).name for m in model.parents])
+        if (
+            self.code_model.options["models_mode"] == "dpg"
+            and model.discriminator_value
+        ):
+            basename += f", discriminator='{model.discriminator_value}'"
         return f"class {model.name}({basename}):{model.pylint_disable}"
 
+    @staticmethod
+    def _escape_dot(s: str):
+        return s.replace(".", "\\\\.")
+
+    @staticmethod
+    def declare_msrest_attribute_map(prop: Property) -> str:
```

Comment:
This is making me think it would be better to have two different serializers. One is the `MsrestModelSerializer` and the other is the `DPGModelSerializer`. They should share vast majority of cvode, but at least for the `attribute_map property` they can have their own definitions

Reply:
basically we don't want to keep doing models mode checks etc

Reply:
Made ModelSerializer be abstract and created two subclass `MsrestModelSerializer` and `DPGModelSerializer` for each model.